pkg/controller/launcher/launcher_controller.go

```diff
@@ -221,12 +221,23 @@ func (r *ReconcileLauncher) Reconcile(request reconcile.Request) (reconcile.Resu
 		if err == nil {
 			clusterConfig.Data["git-providers.yaml"] = string(d)
 		}
+
+		data["launcher.missioncontrol.openshift.password"] = ""
+		data["launcher.missioncontrol.openshift.username"] = ""
+
 		if instance.Spec.OAuth.KeycloakURL != "" {
 			data["launcher.keycloak.url"] = instance.Spec.OAuth.KeycloakURL
 			data["launcher.keycloak.realm"] = instance.Spec.OAuth.KeycloakRealm
 			data["launcher.keycloak.client.id"] = instance.Spec.OAuth.KeycloakClientID
 		} else {
-			data["launcher.oauth.openshift.url"] = instance.Spec.OpenShift.ConsoleURL + "/oauth/authorize"
+			if instance.Spec.OAuth.OauthURL != "" {
+				data["launcher.oauth.openshift.url"] = instance.Spec.OpenShift.ConsoleURL + "/oauth/authorize"
```

Comment:
This looks weird: if the OAuth URL is not empty use the console URL + /oauth/authorize?

Reply:
yep that worked on openshift v3

Reply:
But shouldn't the condition be reverted? If *NOT* empty then use this value otherwise use console +/oauth/authorize ?